stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt

```diff
@@ -58,13 +57,38 @@ class ChannelListFragment : Fragment() {
 
             setLoadingView(loadingView, FrameLayout.LayoutParams(MATCH_PARENT, MATCH_PARENT))
 
-            setChannelClickListener {
+            setChannelItemClickListener {
                 requireActivity().findNavController(R.id.hostFragmentContainer)
                     .navigateSafely(HomeFragmentDirections.actionOpenChat(it.cid, null))
             }
 
-            setDeleteListener {
-                ChatLogger.instance.logD(it.name, "delete me!")
+            setChannelDeleteClickListener { channel ->
+                ChatInfoDeleteChannelDialogFragment
+                    .newInstance()
+                    .apply {
+                        deleteChannelListener = ChatInfoDeleteChannelDialogFragment.ChatInfoDeleteChannelListener {
+                            // should clear history, leave & hide?
```

Comment:
Did you figure out it? Or is it kind of TODO?

Reply:
Sort of a TODO. Not sure what the appropriate order of events is. Backend functionality for leave is still a WIP. Clear history was just sort of me wondering. I think deleting channel might be a little more complicated for group scenarios, and we're still trying to figure out what the "correct" flow is. I was told I should just emulate what Rafal did in ChatInfo, so it only hides for now.